src/AutoSubscriptions.ts

```diff
@@ -403,3 +405,27 @@ export function warnIfAutoSubscribeEnabled<T extends Function>(target: InstanceT
 
     return descriptor;
 }
+
+const autoSubscribeHookHandler = {
+    masterVal: 0,
+    handle(self: any, store: StoreBase, key: string) {
+        const [ _, setter ] = useState(this.masterVal);
+        useEffect(() => {
+            const token = store.subscribe(() => {
+                const val = ++this.masterVal;
+                if (this.masterVal > 1000000000) {
+                    this.masterVal = 0;
+                }
```

Comment:
No need to do all this fancy stuff. Just pass `{}` into the setter and it'll fail reference checks every time thus triggering a re-render.